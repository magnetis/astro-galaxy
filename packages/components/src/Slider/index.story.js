import React, { useState } from 'react';
import { storiesOf } from '@storybook/react';
import Slider from './index.js';

storiesOf('Slider', module)
  .add('default', () => {
    const SliderDefault = props => {
      const [value, setValue] = useState(50);

      return (
        <Slider
          id="slider-default"
          min={-50}
          max={150}
          value={value}
          label="Slider"
          counterLabel={`${value} coins`}
          onChange={e => setValue(e.currentTarget.value)}
          {...props}
        />
      );
    };
    return <SliderDefault />;
  })
  .add('disabled', () => (
    <Slider id="slider-disabled" label="Slider" counterLabel="0 coins" disabled />
  ))
  .add('full fill', () => {
    const SliderFullFill = props => {
      const [value, setValue] = useState(50);

      return (
        <Slider
          id="slider-full-fill"
          min={-50}
          max={150}
          value={value}
          label="Slider"
          counterLabel={`${value} coins`}
          fullFill
          onChange={e => setValue(e.currentTarget.value)}
          {...props}
        />
      );
    };
    return <SliderFullFill />;
  })

  .add('only label', () => {
    const SliderOnlyLabel = props => {
      const [value, setValue] = useState(25);
      return (
        <Slider
          value={value}
          label="Slider"
          id="slider-only-label"
          min={0}
          max={50}
          onChange={e => setValue(e.currentTarget.value)}
          {...props}
        />
      );
    };
    return <SliderOnlyLabel />;
  })
  .add('only counter', () => {
    const SliderOnlyCounter = props => {
      const [value, setValue] = useState(25);
      return (
        <Slider
          value={value}
          counterLabel={`${value} coins`}
          id="slider-only-counter"
          min={0}
          max={50}
          onChange={e => setValue(e.currentTarget.value)}
          {...props}
        />
      );
    };
    return <SliderOnlyCounter />;
  })
  .add('without label and counter', () => {
    const OnlySlider = props => {
      const [value, setValue] = useState(25);
      return (
        <Slider
          value={value}
          min={20}
          max={30}
          onChange={e => setValue(e.currentTarget.value)}
          {...props}
        />
      );
    };
    return <OnlySlider />;
  });
