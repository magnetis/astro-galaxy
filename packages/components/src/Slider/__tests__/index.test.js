import React from 'react';

import Slider from '../index';

describe('Slider', () => {
  it('should render', () => {
    const json = rendererCreateWithTheme(<Slider />).toJSON();
    expect(json).toMatchSnapshot();
  });
});
