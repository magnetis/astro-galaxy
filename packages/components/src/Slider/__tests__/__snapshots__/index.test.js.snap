// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider reders disabled Slider 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 20px;
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  color: #e9eef2;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  cursor: not-allowed;
}

.c1 {
  margin-left: auto;
  font-weight: 600;
}

.c2 {
  width: 100%;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c2:disabled {
  cursor: not-allowed;
  background-color: transparent;
}

.c2::-webkit-slider-runnable-track {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 5px;
  background-color: #afbec9;
  background-image: linear-gradient( to right, #159ce4, 50.000%, transparent 50.000%);
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2:disabled-webkit-slider-runnable-track {
  cursor: not-allowed;
  background-color: #e9eef2;
  background-image: none;
}

.c2::-webkit-slider-thumb {
  width: 28px;
  height: 28px;
  cursor: pointer;
  background-color: #ffffff;
  border-radius: 100%;
  box-shadow: 0 1px 4px 0 rgba(135,135,135,0.5);
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
}

.c2:not(:disabled)::-webkit-slider-thumb:hover,
.c2:focus::-webkit-slider-thumb {
  width: 36px;
  height: 36px;
}

.c2:not(:disabled)::-webkit-slider-thumb:active {
  width: 40px;
  height: 40px;
}

.c2:disabled::-webkit-slider-thumb {
  display: none;
  visibility: hidden;
}

.c2::-moz-range-track {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 5px;
  background-color: #afbec9;
  background-image: linear-gradient( to right, #159ce4, 50.000%, transparent 50.000%);
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2:disabled-moz-range-track {
  cursor: not-allowed;
  background-color: #e9eef2;
  background-image: none;
}

.c2::-moz-range-thumb {
  width: 28px;
  height: 28px;
  cursor: pointer;
  background-color: #ffffff;
  border-radius: 100%;
  box-shadow: 0 1px 4px 0 rgba(135,135,135,0.5);
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
}

.c2:not(:disabled)::-moz-range-thumb:hover,
.c2:focus::-moz-range-thumb {
  width: 36px;
  height: 36px;
}

.c2:not(:disabled)::-moz-range-thumb:active {
  width: 40px;
  height: 40px;
}

.c2:disabled::-moz-range-thumb {
  display: none;
  visibility: hidden;
}

.c2::-ms-track {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 5px;
  background-color: #afbec9;
  background-image: linear-gradient( to right, #159ce4, 50.000%, transparent 50.000%);
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2:disabled-ms-track {
  cursor: not-allowed;
  background-color: #e9eef2;
  background-image: none;
}

.c2::-ms-thumb {
  width: 28px;
  height: 28px;
  cursor: pointer;
  background-color: #ffffff;
  border-radius: 100%;
  box-shadow: 0 1px 4px 0 rgba(135,135,135,0.5);
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
}

.c2:not(:disabled)::-ms-thumb:hover,
.c2:focus::-ms-thumb {
  width: 36px;
  height: 36px;
}

.c2:not(:disabled)::-ms-thumb:active {
  width: 40px;
  height: 40px;
}

.c2:disabled::-ms-thumb {
  display: none;
  visibility: hidden;
}

<div>
  <label
    className="c0"
    disabled={true}
    htmlFor="slider-default"
  >
    <span>
      Slider
    </span>
    <span
      className="c1"
    >
      50 coins
    </span>
  </label>
  <input
    className="c2"
    disabled={true}
    id="slider-default"
    label="Slider"
    max={150}
    min={-50}
    onChange={[Function]}
    type="range"
    value={50}
  />
</div>
`;

exports[`Slider renders Slider with correct ID 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 20px;
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  color: #32383c;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

.c1 {
  margin-left: auto;
  font-weight: 600;
}

.c2 {
  width: 100%;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c2:disabled {
  cursor: not-allowed;
  background-color: transparent;
}

.c2::-webkit-slider-runnable-track {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 5px;
  background-color: #afbec9;
  background-image: linear-gradient( to right, #159ce4, 50.000%, transparent 50.000%);
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2:disabled-webkit-slider-runnable-track {
  cursor: not-allowed;
  background-color: #e9eef2;
  background-image: none;
}

.c2::-webkit-slider-thumb {
  width: 28px;
  height: 28px;
  cursor: pointer;
  background-color: #ffffff;
  border-radius: 100%;
  box-shadow: 0 1px 4px 0 rgba(135,135,135,0.5);
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
}

.c2:not(:disabled)::-webkit-slider-thumb:hover,
.c2:focus::-webkit-slider-thumb {
  width: 36px;
  height: 36px;
}

.c2:not(:disabled)::-webkit-slider-thumb:active {
  width: 40px;
  height: 40px;
}

.c2:disabled::-webkit-slider-thumb {
  display: none;
  visibility: hidden;
}

.c2::-moz-range-track {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 5px;
  background-color: #afbec9;
  background-image: linear-gradient( to right, #159ce4, 50.000%, transparent 50.000%);
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2:disabled-moz-range-track {
  cursor: not-allowed;
  background-color: #e9eef2;
  background-image: none;
}

.c2::-moz-range-thumb {
  width: 28px;
  height: 28px;
  cursor: pointer;
  background-color: #ffffff;
  border-radius: 100%;
  box-shadow: 0 1px 4px 0 rgba(135,135,135,0.5);
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
}

.c2:not(:disabled)::-moz-range-thumb:hover,
.c2:focus::-moz-range-thumb {
  width: 36px;
  height: 36px;
}

.c2:not(:disabled)::-moz-range-thumb:active {
  width: 40px;
  height: 40px;
}

.c2:disabled::-moz-range-thumb {
  display: none;
  visibility: hidden;
}

.c2::-ms-track {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 5px;
  background-color: #afbec9;
  background-image: linear-gradient( to right, #159ce4, 50.000%, transparent 50.000%);
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2:disabled-ms-track {
  cursor: not-allowed;
  background-color: #e9eef2;
  background-image: none;
}

.c2::-ms-thumb {
  width: 28px;
  height: 28px;
  cursor: pointer;
  background-color: #ffffff;
  border-radius: 100%;
  box-shadow: 0 1px 4px 0 rgba(135,135,135,0.5);
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
}

.c2:not(:disabled)::-ms-thumb:hover,
.c2:focus::-ms-thumb {
  width: 36px;
  height: 36px;
}

.c2:not(:disabled)::-ms-thumb:active {
  width: 40px;
  height: 40px;
}

.c2:disabled::-ms-thumb {
  display: none;
  visibility: hidden;
}

<div>
  <label
    className="c0"
    disabled={false}
    htmlFor="slider-id"
  >
    <span>
      Slider
    </span>
    <span
      className="c1"
    >
      50 coins
    </span>
  </label>
  <input
    className="c2"
    disabled={false}
    id="slider-id"
    label="Slider"
    max={150}
    min={-50}
    onChange={[Function]}
    type="range"
    value={50}
  />
</div>
`;

exports[`Slider renders default slider 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 20px;
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  color: #32383c;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

.c1 {
  margin-left: auto;
  font-weight: 600;
}

.c2 {
  width: 100%;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c2:disabled {
  cursor: not-allowed;
  background-color: transparent;
}

.c2::-webkit-slider-runnable-track {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 5px;
  background-color: #afbec9;
  background-image: linear-gradient( to right, #159ce4, 50.000%, transparent 50.000%);
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2:disabled-webkit-slider-runnable-track {
  cursor: not-allowed;
  background-color: #e9eef2;
  background-image: none;
}

.c2::-webkit-slider-thumb {
  width: 28px;
  height: 28px;
  cursor: pointer;
  background-color: #ffffff;
  border-radius: 100%;
  box-shadow: 0 1px 4px 0 rgba(135,135,135,0.5);
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
}

.c2:not(:disabled)::-webkit-slider-thumb:hover,
.c2:focus::-webkit-slider-thumb {
  width: 36px;
  height: 36px;
}

.c2:not(:disabled)::-webkit-slider-thumb:active {
  width: 40px;
  height: 40px;
}

.c2:disabled::-webkit-slider-thumb {
  display: none;
  visibility: hidden;
}

.c2::-moz-range-track {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 5px;
  background-color: #afbec9;
  background-image: linear-gradient( to right, #159ce4, 50.000%, transparent 50.000%);
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2:disabled-moz-range-track {
  cursor: not-allowed;
  background-color: #e9eef2;
  background-image: none;
}

.c2::-moz-range-thumb {
  width: 28px;
  height: 28px;
  cursor: pointer;
  background-color: #ffffff;
  border-radius: 100%;
  box-shadow: 0 1px 4px 0 rgba(135,135,135,0.5);
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
}

.c2:not(:disabled)::-moz-range-thumb:hover,
.c2:focus::-moz-range-thumb {
  width: 36px;
  height: 36px;
}

.c2:not(:disabled)::-moz-range-thumb:active {
  width: 40px;
  height: 40px;
}

.c2:disabled::-moz-range-thumb {
  display: none;
  visibility: hidden;
}

.c2::-ms-track {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 5px;
  background-color: #afbec9;
  background-image: linear-gradient( to right, #159ce4, 50.000%, transparent 50.000%);
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2:disabled-ms-track {
  cursor: not-allowed;
  background-color: #e9eef2;
  background-image: none;
}

.c2::-ms-thumb {
  width: 28px;
  height: 28px;
  cursor: pointer;
  background-color: #ffffff;
  border-radius: 100%;
  box-shadow: 0 1px 4px 0 rgba(135,135,135,0.5);
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
}

.c2:not(:disabled)::-ms-thumb:hover,
.c2:focus::-ms-thumb {
  width: 36px;
  height: 36px;
}

.c2:not(:disabled)::-ms-thumb:active {
  width: 40px;
  height: 40px;
}

.c2:disabled::-ms-thumb {
  display: none;
  visibility: hidden;
}

<div>
  <label
    className="c0"
    disabled={false}
    htmlFor="slider-default"
  >
    <span>
      Slider
    </span>
    <span
      className="c1"
    >
      50 coins
    </span>
  </label>
  <input
    className="c2"
    disabled={false}
    id="slider-default"
    label="Slider"
    max={150}
    min={-50}
    onChange={[MockFunction]}
    type="range"
    value={50}
  />
</div>
`;

exports[`Slider renders with counter only 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 20px;
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  color: #32383c;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

.c1 {
  margin-left: auto;
  font-weight: 600;
}

.c2 {
  width: 100%;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c2:disabled {
  cursor: not-allowed;
  background-color: transparent;
}

.c2::-webkit-slider-runnable-track {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 5px;
  background-color: #afbec9;
  background-image: linear-gradient( to right, #159ce4, 50.000%, transparent 50.000%);
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2:disabled-webkit-slider-runnable-track {
  cursor: not-allowed;
  background-color: #e9eef2;
  background-image: none;
}

.c2::-webkit-slider-thumb {
  width: 28px;
  height: 28px;
  cursor: pointer;
  background-color: #ffffff;
  border-radius: 100%;
  box-shadow: 0 1px 4px 0 rgba(135,135,135,0.5);
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
}

.c2:not(:disabled)::-webkit-slider-thumb:hover,
.c2:focus::-webkit-slider-thumb {
  width: 36px;
  height: 36px;
}

.c2:not(:disabled)::-webkit-slider-thumb:active {
  width: 40px;
  height: 40px;
}

.c2:disabled::-webkit-slider-thumb {
  display: none;
  visibility: hidden;
}

.c2::-moz-range-track {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 5px;
  background-color: #afbec9;
  background-image: linear-gradient( to right, #159ce4, 50.000%, transparent 50.000%);
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2:disabled-moz-range-track {
  cursor: not-allowed;
  background-color: #e9eef2;
  background-image: none;
}

.c2::-moz-range-thumb {
  width: 28px;
  height: 28px;
  cursor: pointer;
  background-color: #ffffff;
  border-radius: 100%;
  box-shadow: 0 1px 4px 0 rgba(135,135,135,0.5);
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
}

.c2:not(:disabled)::-moz-range-thumb:hover,
.c2:focus::-moz-range-thumb {
  width: 36px;
  height: 36px;
}

.c2:not(:disabled)::-moz-range-thumb:active {
  width: 40px;
  height: 40px;
}

.c2:disabled::-moz-range-thumb {
  display: none;
  visibility: hidden;
}

.c2::-ms-track {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 5px;
  background-color: #afbec9;
  background-image: linear-gradient( to right, #159ce4, 50.000%, transparent 50.000%);
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2:disabled-ms-track {
  cursor: not-allowed;
  background-color: #e9eef2;
  background-image: none;
}

.c2::-ms-thumb {
  width: 28px;
  height: 28px;
  cursor: pointer;
  background-color: #ffffff;
  border-radius: 100%;
  box-shadow: 0 1px 4px 0 rgba(135,135,135,0.5);
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
}

.c2:not(:disabled)::-ms-thumb:hover,
.c2:focus::-ms-thumb {
  width: 36px;
  height: 36px;
}

.c2:not(:disabled)::-ms-thumb:active {
  width: 40px;
  height: 40px;
}

.c2:disabled::-ms-thumb {
  display: none;
  visibility: hidden;
}

<div>
  <label
    className="c0"
    disabled={false}
    htmlFor="slider-default"
  >
    <span
      className="c1"
    >
      30 coins
    </span>
  </label>
  <input
    className="c2"
    disabled={false}
    id="slider-default"
    label={null}
    max={150}
    min={-50}
    onChange={[Function]}
    type="range"
    value={50}
  />
</div>
`;

exports[`Slider renders with counter only 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 20px;
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  color: #32383c;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

.c1 {
  width: 100%;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c1:disabled {
  cursor: not-allowed;
  background-color: transparent;
}

.c1::-webkit-slider-runnable-track {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 5px;
  background-color: #afbec9;
  background-image: linear-gradient( to right, #159ce4, 50.000%, transparent 50.000%);
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1:disabled-webkit-slider-runnable-track {
  cursor: not-allowed;
  background-color: #e9eef2;
  background-image: none;
}

.c1::-webkit-slider-thumb {
  width: 28px;
  height: 28px;
  cursor: pointer;
  background-color: #ffffff;
  border-radius: 100%;
  box-shadow: 0 1px 4px 0 rgba(135,135,135,0.5);
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
}

.c1:not(:disabled)::-webkit-slider-thumb:hover,
.c1:focus::-webkit-slider-thumb {
  width: 36px;
  height: 36px;
}

.c1:not(:disabled)::-webkit-slider-thumb:active {
  width: 40px;
  height: 40px;
}

.c1:disabled::-webkit-slider-thumb {
  display: none;
  visibility: hidden;
}

.c1::-moz-range-track {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 5px;
  background-color: #afbec9;
  background-image: linear-gradient( to right, #159ce4, 50.000%, transparent 50.000%);
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1:disabled-moz-range-track {
  cursor: not-allowed;
  background-color: #e9eef2;
  background-image: none;
}

.c1::-moz-range-thumb {
  width: 28px;
  height: 28px;
  cursor: pointer;
  background-color: #ffffff;
  border-radius: 100%;
  box-shadow: 0 1px 4px 0 rgba(135,135,135,0.5);
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
}

.c1:not(:disabled)::-moz-range-thumb:hover,
.c1:focus::-moz-range-thumb {
  width: 36px;
  height: 36px;
}

.c1:not(:disabled)::-moz-range-thumb:active {
  width: 40px;
  height: 40px;
}

.c1:disabled::-moz-range-thumb {
  display: none;
  visibility: hidden;
}

.c1::-ms-track {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 5px;
  background-color: #afbec9;
  background-image: linear-gradient( to right, #159ce4, 50.000%, transparent 50.000%);
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1:disabled-ms-track {
  cursor: not-allowed;
  background-color: #e9eef2;
  background-image: none;
}

.c1::-ms-thumb {
  width: 28px;
  height: 28px;
  cursor: pointer;
  background-color: #ffffff;
  border-radius: 100%;
  box-shadow: 0 1px 4px 0 rgba(135,135,135,0.5);
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
}

.c1:not(:disabled)::-ms-thumb:hover,
.c1:focus::-ms-thumb {
  width: 36px;
  height: 36px;
}

.c1:not(:disabled)::-ms-thumb:active {
  width: 40px;
  height: 40px;
}

.c1:disabled::-ms-thumb {
  display: none;
  visibility: hidden;
}

<div>
  <label
    className="c0"
    disabled={false}
    htmlFor="slider-default"
  />
  <input
    className="c1"
    disabled={false}
    id="slider-default"
    label={null}
    max={150}
    min={-50}
    onChange={[Function]}
    type="range"
    value={50}
  />
</div>
`;

exports[`Slider renders with label only 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 20px;
  font: normal normal 400 normal 16px/1.5 Lato,sans-serif;
  color: #32383c;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

.c1 {
  width: 100%;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c1:disabled {
  cursor: not-allowed;
  background-color: transparent;
}

.c1::-webkit-slider-runnable-track {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 5px;
  background-color: #afbec9;
  background-image: linear-gradient( to right, #159ce4, 50.000%, transparent 50.000%);
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1:disabled-webkit-slider-runnable-track {
  cursor: not-allowed;
  background-color: #e9eef2;
  background-image: none;
}

.c1::-webkit-slider-thumb {
  width: 28px;
  height: 28px;
  cursor: pointer;
  background-color: #ffffff;
  border-radius: 100%;
  box-shadow: 0 1px 4px 0 rgba(135,135,135,0.5);
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
}

.c1:not(:disabled)::-webkit-slider-thumb:hover,
.c1:focus::-webkit-slider-thumb {
  width: 36px;
  height: 36px;
}

.c1:not(:disabled)::-webkit-slider-thumb:active {
  width: 40px;
  height: 40px;
}

.c1:disabled::-webkit-slider-thumb {
  display: none;
  visibility: hidden;
}

.c1::-moz-range-track {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 5px;
  background-color: #afbec9;
  background-image: linear-gradient( to right, #159ce4, 50.000%, transparent 50.000%);
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1:disabled-moz-range-track {
  cursor: not-allowed;
  background-color: #e9eef2;
  background-image: none;
}

.c1::-moz-range-thumb {
  width: 28px;
  height: 28px;
  cursor: pointer;
  background-color: #ffffff;
  border-radius: 100%;
  box-shadow: 0 1px 4px 0 rgba(135,135,135,0.5);
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
}

.c1:not(:disabled)::-moz-range-thumb:hover,
.c1:focus::-moz-range-thumb {
  width: 36px;
  height: 36px;
}

.c1:not(:disabled)::-moz-range-thumb:active {
  width: 40px;
  height: 40px;
}

.c1:disabled::-moz-range-thumb {
  display: none;
  visibility: hidden;
}

.c1::-ms-track {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 5px;
  background-color: #afbec9;
  background-image: linear-gradient( to right, #159ce4, 50.000%, transparent 50.000%);
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1:disabled-ms-track {
  cursor: not-allowed;
  background-color: #e9eef2;
  background-image: none;
}

.c1::-ms-thumb {
  width: 28px;
  height: 28px;
  cursor: pointer;
  background-color: #ffffff;
  border-radius: 100%;
  box-shadow: 0 1px 4px 0 rgba(135,135,135,0.5);
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
}

.c1:not(:disabled)::-ms-thumb:hover,
.c1:focus::-ms-thumb {
  width: 36px;
  height: 36px;
}

.c1:not(:disabled)::-ms-thumb:active {
  width: 40px;
  height: 40px;
}

.c1:disabled::-ms-thumb {
  display: none;
  visibility: hidden;
}

<div>
  <label
    className="c0"
    disabled={false}
    htmlFor="slider-default"
  >
    <span>
      Slider label
    </span>
  </label>
  <input
    className="c1"
    disabled={false}
    id="slider-default"
    label="Slider label"
    max={150}
    min={-50}
    onChange={[Function]}
    type="range"
    value={50}
  />
</div>
`;
